main.py

```diff
@@ -558,6 +562,21 @@ def main(cl_arguments):
     if args.do_full_eval:
         log.info("Evaluating...")
         splits_to_write = evaluate.parse_write_preds_arg(args.write_preds)
+        diagnostic_tasks = []
+        target_tasks_to_remove = []
+        for index, task in enumerate(target_tasks):
+            if task.name in task_modules.ALL_DIAGNOSTICS:
+                diagnostic_tasks.append(task)
+                target_tasks_to_remove.append(index)
+        # To avoid evaluating twice on a diagnostic task.
+        for index in sorted(target_tasks_to_remove, reverse=True):
```

Comment:
Why sort and reverse? Why not just del target_tasks(index) in the previous loop?

Reply:
Because if you delete from first to last, then the indices will get messed up (remove index 0, then all the proceeding indices need to be shuffled down by 1).

Reply:
Ah, fair point. Another easy alternative: target_tasks.remove(task) in the initial loop, with no indexing at all.